types/mongodb/index.d.ts

```diff
@@ -1217,7 +1217,7 @@ type RegExpString<T> = T extends string ? (RegExp | T) : T;
 
 export type FilterQuery<T> = {
     [P in keyof T]?: RegExpString<T[P]> | QuerySelector<RegExpString<T[P]>>
-} & RootQuerySelector<T>;
+} | RootQuerySelector<T>;
```

Comment:
Using `|` instead of `&` here makes this type like a simple `object` type because of `[key: string]: any` at the end of `RootQuerySelector`.